sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/extensions/gcp/util/GcsUtil.java

```diff
@@ -427,23 +449,29 @@ public WritableByteChannel create(GcsPath path, String type) throws IOException
    */
   public WritableByteChannel create(GcsPath path, String type, Integer uploadBufferSizeBytes)
       throws IOException {
-    GoogleCloudStorageWriteChannel channel =
-        new GoogleCloudStorageWriteChannel(
-            executorService,
-            storageClient,
-            new ClientRequestHelper<>(),
-            path.getBucket(),
-            path.getObject(),
-            type,
-            /* kmsKeyName= */ null,
-            AsyncWriteChannelOptions.newBuilder().build(),
-            new ObjectWriteConditions(),
-            Collections.emptyMap());
-    if (uploadBufferSizeBytes != null) {
-      channel.setUploadBufferSize(uploadBufferSizeBytes);
+    if (uploadBufferSizeBytes == null) {
```

Comment:
Was this addressed ? I see this being used in a number of places and probably users can set mime type themselves since this is a pipeline option. Can this be a breaking change for some users (or some applications of Beam) ?

For example,
https://github.com/apache/beam/blob/35da90a94953597e9e5676e1e1e70f27d2a8f064/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryRowWriter.java#L58

Reply:
I think this can be address by passing `CreateObjectOptions` to `googleCloudStorage.create`. `CreateObjectOptions` has a way to configure `ContentType`.